pyrail/models.py

```
@@ -330,16 +303,30 @@ class ConnectionDetails(DataClassORJSONMixin):
     departure: ConnectionDeparture  # Departure details
     arrival: ConnectionArrival  # Arrival details
     duration: int  # Duration of the connection in minutes
-    remarks: Remarks  # Remarks for the connection
-    alerts: Alerts  # Alerts for the connection
-    vias: Vias | None = field(default=None)  # Vias information
+    remarks: list[Remark] = field(default_factory=list)  # List of remarks for the connection
+    alerts: list[Alert] = field(default_factory=list)  # List of alerts for the connection
+    vias: list[Via] | None = field(default=None)  # List of via details
+
+    @classmethod
+    def __pre_deserialize__(cls, d: dict[Any, Any]) -> dict[Any, Any]:
```

Ruff (0.8.2)

Missing docstring in magic method

(D105)